cvat/apps/lambda_manager/views.py

```diff
@@ -599,6 +667,96 @@ def reset(self):
+                # Invalid label provided
+                return None
+
+            if anno.get('attributes'):
+                attrs = [{
+                    'spec_id': label['attributes'][attr['name']],
+                    'value': attr['value']
+                } for attr in anno.get('attributes') if attr['name'] in label['attributes']]
+            else:
+                attrs = []
+
+            if anno["type"].lower() == "tag":
+                return {
+                    "frame": frame,
+                    "label_id": label['id'],
+                    "source": "auto",
+                    "attributes": attrs,
+                    "group": None,
+                }
+            else:
+                shape = {
+                    "frame": frame,
+                    "label_id": label['id'],
+                    "source": "auto",
+                    "attributes": attrs,
+                    "group": anno["group_id"] if "group_id" in anno else None,
+                    "type": anno["type"],
+                    "occluded": False,
+                    "points": anno.get("mask", []) if anno["type"] == "mask" else anno.get("points", []),
+                    "z_order": 0,
+                }
+
+                if shape["type"] in ("rectangle", "ellipse"):
+                    shape["rotation"] = anno.get("rotation", 0)
+
+                if anno["type"] == "mask" and "points" in anno and conv_mask_to_poly:
+                    shape["type"] = "polygon"
+                    shape["points"] = anno["points"]
+                elif anno["type"] == "mask":
+                    [xtl, ytl, xbr, ybr] = shape["points"][-4:]
+                    cut_points = shape["points"][:-4]
+                    rle = mask_tools.mask_to_rle(np.array(cut_points))["counts"].tolist()
+                    rle.extend([xtl, ytl, xbr, ybr])
+                    shape["points"] = rle
+
+                if shape["type"] == "skeleton":
+                    parsed_elements = list(map(lambda x: parse_anno(x, label['sublabels']), anno["elements"]))
+
+                    # find a center to set position of lack points
+                    center = [0, 0]
+                    for element in parsed_elements:
+                        center[0] += element["points"][0]
+                        center[1] += element["points"][1]
+                    center[0] /= len(parsed_elements) or 1
+                    center[1] /= len(parsed_elements) or 1
+
+                    def _map(sublabel):
+                        sublabel_name, sublabel_body = sublabel
+                        parsed_element = next(filter(lambda x: x['label_id'] == sublabel_body['id'], parsed_elements), None)
+                        not_parsed_element = next(filter(lambda x: x['label'] == sublabel_name, anno["elements"]), None)
+
+                        if parsed_element:
+                            parsed_element.update({
+                                "outside": not_parsed_element.get('outside', False),
```

Comment:
This seems weird. Why not set `shape['outside']` in the body of `parse_anno`, so that `parsed_element` would already have this information?